pkg/utils/oci/helmregistry.go

```diff
@@ -52,6 +58,18 @@ func (r *HelmRegistry) Pull(ctx context.Context, oci *gardencorev1.OCIRepository
 		remote.WithContext(ctx),
 	}
 
+	if oci.PullSecretRef != nil {
+		secret := &corev1.Secret{}
+		key := client.ObjectKey{Namespace: v1beta1constants.GardenNamespace, Name: oci.PullSecretRef.Name}
+		if err := r.client.Get(ctx, key, secret); err != nil {
+			return nil, fmt.Errorf("failed to get pull secret %s: %w", key, err)
+		}
+		if secret.Data == nil || secret.Data[corev1.DockerConfigJsonKey] == nil {
+			return nil, fmt.Errorf("pull secret %s is missing the data key %s", key, corev1.DockerConfigJsonKey)
```

Comment:
Nit: The typical style is
```suggestion
		secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: v1beta1constants.GardenNamespace, Name: oci.PullSecretRef.Name}}
		if err := r.client.Get(ctx, client.ObjectKeyFromObject(secret), secret); err != nil {
			return nil, fmt.Errorf("failed to get pull secret %s: %w", client.ObjectKeyFromObject(secret), err)
}
		if secret.Data == nil || secret.Data[corev1.DockerConfigJsonKey] == nil {
			return nil, fmt.Errorf("pull secret %s is missing the data key %s",client.ObjectKeyFromObject(secret), corev1.DockerConfigJsonKey)
```
```